### PR TITLE
Add some draft guidelines for modules `meta.yml`

### PR DESCRIPTION
**Pull request by jfy133**
This is a draft and open for discussion.

I've based it on what I normally comment on PRs based on my 'mental model' of what it should like (based on the original modules when the whole thing was being set up)

<a href="https://gitpod.io/#https://github.com/nf-core/nf-co.re/pull/1482"></a>

### CODE REVIEW

File: markdown/developers/modules.md [hunk 1]
```diff
@@ -518,6 +518,35 @@ process {
 
 5. Each output file SHOULD be emitted in it's own channel, along with the `meta` map if provided ( the exception is the versions.yml ).
 
+### Documentation
+
+1. Each module MUST have a `meta.yaml` in the same directory as the `main.nf` of the module itself.
+
+2. Keywords SHOULD be sufficient to make the module findable through research domain, data types, and tool function keywords
+
+   - Keywords MUST NOT just be solely of the (sub)tool name
```
Comment (FriederikeHanssen) [inline markdown/developers/modules.md:527]: not for this PR, but can/should we collect keywords somewhere, so one can check what is out there? Some form of controlled words would be awesome at some point

Comment (FriederikeHanssen) [reply markdown/developers/modules.md:527]: hm I guess you can check the modules webpage, but not very obvious maybe.

Comment (jfy133) [reply markdown/developers/modules.md:527]: That's the plan for one of spring cleaning tasks, -  go through them all and try and consolidate/standardise them (controlled words might be tricky though)

**Review by muffato — APPROVED** [review]

File: markdown/developers/modules.md (hunk 1 above)
Comment (FriederikeHanssen) [reply markdown/developers/modules.md:527]: ah right, I remember we talked about this at some point, but forgot the outcome 😆

File: markdown/developers/modules.md [hunk 2]
```diff
@@ -518,6 +518,35 @@ process {
...
+
+3. Keywords MUST be all lower case
+
+4. Input and Output sections of the meta.yaml SHOULD only have entries of input and output channels
+
+5. Tuples input and outputs MUST be split into separate entries
+
+   - i.e., `meta` should be a separate entry to the `file` it is associated with
+
+6. Input/output types MUST only be of the following categories: `string`, `file`, `directory`.
```
Comment (mahesh-panchal) [inline markdown/developers/modules.md:537]: What about number? https://github.com/nf-core/modules/blob/7e89020bbbade6b032c11219b723d11f4a4bbc66/modules/nf-core/gfastats/meta.yml#L32-L34

Comment (jfy133) [reply markdown/developers/modules.md:537]: Agreed! But should it just be generic number? or `integer`/`float` (given I've personally specified file/directory)? We could just have string/number (accounts for both integer/float)/file (both files and directories)

Comment (mahesh-panchal) [reply markdown/developers/modules.md:537]: Integer/float then in that case.

Comment (jfy133) [inline markdown/developers/modules.md:537]:
```suggestion
6. Input/output types MUST only be of the following categories: `file`, `directory`, `string, `integer`, `float` .
```

File: markdown/developers/modules.md [hunk 3]
```diff
@@ -518,6 +518,35 @@ process {
...
+
+2. Keywords SHOULD be sufficient to make the module findable through research domain, data types, and tool function keywords
+
+   - Keywords MUST NOT just be solely of the (sub)tool name
+
+3. Keywords MUST be all lower case
+
+4. Input and Output sections of the meta.yaml SHOULD only have entries of input and output channels
+
+5. Tuples input and outputs MUST be split into separate entries
```
Comment (mahesh-panchal) [inline markdown/developers/modules.md:533]:
```suggestion
5. Input and output tuples MUST be split into separate entries
```

File: markdown/developers/modules.md [hunk 4]
```diff
@@ -518,6 +518,35 @@ process {
...
+
+   - There should be a one-to-one relationship between the module and the `meta.yaml`.
+   - Input/output entries MUST NOT combine multiple output channels
+
+8. Input/output descriptions SHOULD be descriptive of the contents of file
+
+   - i.e., not just 'A TSV file'
+
+9. Input/output patterns (if present) MUST follow a [Java glob pattern](https://docs.oracle.com/javase/tutorial/essential/io/fileOps.html#glob)
+
```
Comment (mahesh-panchal) [inline markdown/developers/modules.md:549]:
```suggestion

10. Input/Output entries should be marked as Mandatory or Optional.
```

**Review by Emiller88 — APPROVED** [review]
LGTM if @mahesh-panchal's changes get committed!

**Review by mahesh-panchal — APPROVED** [review]

**Pull request merged**